trustfall_core/src/graphql_query/query.rs

```diff
@@ -125,49 +125,52 @@ fn try_get_query_root(document: &ExecutableDocument) -> Result<&Positioned<Field
 
     match &document.operations {
         DocumentOperations::Multiple(mult) => {
-            return Err(ParseError::MultipleOperationsInDocument(mult.values().next().unwrap().pos))
+            return if mult.values().len() > 1 {
+                Err(ParseError::MultipleOperationsInDocument(mult.values().next().unwrap().pos))
+            } else {
+                let node = mult.values().next().unwrap();
+                parse_operation_definition(node)
+            }
```

Comment:
I wrote this code really sloppily back when I didn't really know what I was doing in Rust. Would you mind making a couple of improvements along the way?

Could we:
- Replace all `.unwrap()` here with `expect()`
- Use an `if let Some()` to extract the single value instead of `unwrap()/expect()`
- If there is more than one value, report the `pos` of the _second_ operation instead of the first

Reply:
Since we are returning a result couldn't I replace `.unwrap()` with a try operator? i.e. `?`

Reply:
Ope disregard - most of these return an `Option` 😆